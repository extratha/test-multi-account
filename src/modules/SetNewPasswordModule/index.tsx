"use client";

import { SubmitButtonStyle } from "@/components/Button/styled";
import { API } from "@/constant/api";
import { COOKIE } from "@/constant/constant";
import { webPaths } from "@/constant/webPaths";
import useTranslation from "@/locales/useLocale";
import { CustomTextField } from "@/modules/LoginModule/styled";
import { usePageLoadingStore } from "@/store";
import useToastStore from "@/store/useToastStore";
import axiosInstance, { AxiosInstance } from "@/utils/axios";
import axiosPublicInstance from "@/utils/axios/login";
import VisibilityIcon from "@mui/icons-material/Visibility";
import VisibilityOffIcon from "@mui/icons-material/VisibilityOff";
import { IconButton, Stack, styled, Typography, useTheme } from "@mui/material";
import { deleteCookie, getCookie, setCookie } from "cookies-next";
import { useRouter } from "next/navigation";
import { useState } from "react";
import { Controller, SubmitHandler, useForm } from "react-hook-form";

interface SetNewPasswordForm {
  newPassword: string;
  confirmNewPassword: string;
}
const fieldName: Record<string, keyof SetNewPasswordForm> = {
  NEW_PASSWORD: "newPassword",
  CONFIRM_NEW_PASSWORD: "confirmNewPassword",
};

const passwordRegex = /^(?=.*[A-Z])(?=.*[a-z])(?=.*\d)(?=.*\W).{8,}$/;

const SetNewPasswordWrapper = styled(Stack)(({ theme }) => ({
  color: theme.palette.text.medium,
  width: "66%",
  margin: "auto",
}));

const SetNewPasswordModule = () => {
  const { translation } = useTranslation();
  const theme = useTheme();
  const router = useRouter();
  const { setToastOpen } = useToastStore();
  const [errorMessage, setErrorMessage] = useState<string>("");
  const [showPassword, setShowPassword] = useState<boolean>(false);
  const [showConfirmPassword, setShowConfirmPassword] = useState<boolean>(false);
  const { setPageLoading } = usePageLoadingStore();
  const [newPassword, setNewPassword] = useState<string>("");
  const [confirmNewPassword, setConfirmNewPassword] = useState<string>("");
  const [isDisableSubmit, setIsDisabledSubmit] = useState(true);
  const { handleSubmit, control, getValues, setError } = useForm<SetNewPasswordForm>();
  const resetPasswordToken = getCookie(COOKIE.RESET_PASSWORD_TOKEN);
  const fetcher: AxiosInstance = resetPasswordToken ? axiosPublicInstance : axiosInstance;
  const apiUrl: string = resetPasswordToken ? API.PATH.setNewPassword : API.PATH.changePassword;
  let submitParams = {};
  if (resetPasswordToken) {
    submitParams = {
      passwordResetToken: resetPasswordToken,
      password: newPassword,
      confirmPassword: confirmNewPassword,
    };
  } else {
    submitParams = {
      newPassword: newPassword,
    };
  }
  const onSubmit: SubmitHandler<SetNewPasswordForm> = async () => {
    setPageLoading(true);

    try {
      const response = await fetcher.post(apiUrl, submitParams);
      if (response?.status === 204) {
        setCookie(COOKIE.PASSWORD_CHANGED, true);
        if (resetPasswordToken && resetPasswordToken?.length > 0) {
          deleteCookie(COOKIE.RESET_PASSWORD_TOKEN);
        }
        setToastOpen(true, {
          message: translation("Common.toast.resetPasswordSuccess"),
          severity: "success",
          icon: <div />,
        });
        router.push(webPaths.login);
      }
    } catch (error: any) {
      if (error.message) {
        setErrorMessage(error.message);
      } else {
        setErrorMessage(error.error);
      }
      setPageLoading(false);
    }
  };
  const isValidPassword = (password: string) => {
    if (!passwordRegex.test(password)) {
      return false;
    }
    return true;
  };
  const validateConfirmNewPasswordField = (value: unknown, fieldNameProp: keyof SetNewPasswordForm) => {
    const siblingFieldName =
      fieldNameProp === fieldName.NEW_PASSWORD ? fieldName.CONFIRM_NEW_PASSWORD : fieldName.NEW_PASSWORD;
    const siblingValue = getValues(siblingFieldName);
    setError(fieldNameProp, { type: "", message: "" });
    setIsDisabledSubmit(true);
    if (!value) {
      setError(fieldNameProp, { type: "validate", message: translation("Common.validation.require") });
      return true;
    }
    if (typeof value === "string" && !isValidPassword(value)) {
      setError(fieldNameProp, { type: "validate", message: translation("Common.validation.invalidPasswordFormat") });
      return true;
    }
    const isPasswordMatched = value === siblingValue;
    if (value && siblingValue && !isPasswordMatched) {
      setError(fieldNameProp, { type: "validate", message: translation("Common.validation.passwordIsNotMatch") });
      setError(siblingFieldName, { type: "validate", message: translation("Common.validation.passwordIsNotMatch") });
      return true;
    } else {
      setError(siblingFieldName, { type: "", message: "" });
    }
    if (value && siblingValue) {
      setIsDisabledSubmit(false);
    }
    return false;
  };
  return (
    <SetNewPasswordWrapper>
      <Stack>
        <Typography variant="headerBold" mb={2}>
          {translation("Common.pages.setNewPassword")}
        </Typography>
        <Typography variant="bodySmall" color="text.medium" mb={2}>
          {translation("Common.text.setNewPassword")}
        </Typography>
        <form onSubmit={handleSubmit(onSubmit)} autoFocus>
          <Controller
            name={fieldName.NEW_PASSWORD as keyof SetNewPasswordForm}
            control={control}
            render={({ field, fieldState: { error } }) => (
              <Stack direction="column" spacing={2} mt={1}>
                <CustomTextField
                  {...field}
                  fullWidth
                  autoFocus
                  id={fieldName.NEW_PASSWORD}
                  data-testid={fieldName.NEW_PASSWORD}
                  type={showPassword ? "text" : "password"}
                  placeholder={!newPassword ? "ตั้งค่ารหัสผ่าน" : ""}
                  name={fieldName.NEW_PASSWORD}
                  value={(newPassword as unknown) ?? ""}
                  onChange={(event) => {
                    const value = event?.target?.value;
                    field.onChange(event);
                    setNewPassword(value);
                    validateConfirmNewPasswordField(value, fieldName.NEW_PASSWORD);
                  }}
                  InputProps={{
                    endAdornment: (
                      <a>
                        <IconButton
                          aria-label="Toggle show password"
                          data-testid="button-toggle-show-new-password"
                          onClick={() => setShowPassword((prev) => !prev)}
                        >
                          {showPassword ? (
                            <VisibilityIcon aria-label="Showing password" />
                          ) : (
                            <VisibilityOffIcon aria-label="Hiding password" />
                          )}
                        </IconButton>
                      </a>
                    ),
                  }}
                  InputLabelProps={{
                    shrink: true,
                  }}
                  sx={{
                    marginTop: "2em",
                  }}
                />
                <Typography
                  data-testid="error"
                  variant="labelExtraSmallMedium"
                  mt={0}
                  sx={{
                    textAlign: "left",
                    color: theme.palette.error.main,
                  }}
                >
                  {error?.message || ""}
                </Typography>
              </Stack>
            )}
          />

          <Controller
            name={fieldName.CONFIRM_NEW_PASSWORD as keyof SetNewPasswordForm}
            control={control}
            render={({ field, fieldState: { error } }) => (
              <Stack direction="column" spacing={2} mt={1}>
                <CustomTextField
                  {...field}
                  fullWidth
                  autoFocus
                  name={fieldName.CONFIRM_NEW_PASSWORD}
                  type={showConfirmPassword ? "text" : "password"}
                  id={fieldName.CONFIRM_NEW_PASSWORD}
                  data-testid={fieldName.CONFIRM_NEW_PASSWORD}
                  placeholder={!confirmNewPassword ? "ตั้งรหัสผ่านใหม่อีกครั้ง" : ""}
                  value={(confirmNewPassword as unknown) ?? ""}
                  onChange={(event) => {
                    const value = event?.target?.value;
                    field.onChange(event);
                    validateConfirmNewPasswordField(value, fieldName.CONFIRM_NEW_PASSWORD);
                    setConfirmNewPassword(value);
                  }}
                  InputProps={{
                    endAdornment: (
                      <a>
                        <IconButton
                          data-testid="button-toggle-show-confirm-password"
                          aria-label="Toggle show password"
                          onClick={() => setShowConfirmPassword((prev) => !prev)}
                        >
                          {showConfirmPassword ? (
                            <VisibilityIcon aria-label="Showing password" />
                          ) : (
                            <VisibilityOffIcon aria-label="Hiding password" />
                          )}
                        </IconButton>
                      </a>
                    ),
                  }}
                  InputLabelProps={{
                    shrink: true,
                  }}
                  sx={{
                    marginTop: "2em",
                  }}
                />
                <Typography
                  data-testid="error"
                  variant="labelExtraSmallMedium"
                  sx={{
                    textAlign: "left",
                    color: theme.palette.error.main,
                  }}
                >
                  {error?.message || ""}
                </Typography>
              </Stack>
            )}
          />

          <Typography
            variant="labelExtraSmallMedium"
            sx={{
              textAlign: "left",
              color: theme.palette.error.main,
            }}
          >
            {errorMessage || ""}
          </Typography>

          <Stack mt={1}>
            <Typography variant="labelExtraSmallBold">
              {translation("Common.text.setNewPasswordValidateTitle")}
            </Typography>
            <Typography variant="labelExtraSmall" whiteSpace={"break-spaces"}>
              {translation("Common.text.setNewPasswordValidateMessage")}
            </Typography>
          </Stack>
          <SubmitButtonStyle data-testid="button-set-new-password" type="submit" disabled={isDisableSubmit}>
            {translation("Common.button.setNewPassword")}
          </SubmitButtonStyle>
        </form>
      </Stack>
    </SetNewPasswordWrapper>
  );
};

export default SetNewPasswordModule;
