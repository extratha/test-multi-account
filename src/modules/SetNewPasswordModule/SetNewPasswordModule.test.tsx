import { render } from "@/testUtils/testUtils";
import SetNewPasswordModule from ".";

jest.mock("next/navigation", () => ({
  ...jest.requireActual("next/navigation"),
  useRouter: jest.fn(),
}));

describe("SetNewPasswordModule", () => {
  it("should render correctly", async () => {
    const { asFragment } = render(<SetNewPasswordModule />);
    expect(asFragment()).toMatchSnapshot();
  });
});
