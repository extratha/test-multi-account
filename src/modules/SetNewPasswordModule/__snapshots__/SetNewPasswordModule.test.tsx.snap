// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SetNewPasswordModule should render correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 100vw;
  height: 100vh;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.emotion-2 {
  position: relative;
  margin: 40vh auto 0;
}

.emotion-3 {
  position: relative;
  margin: auto auto 3rem;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  background-color: #fff;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 66%;
  margin: auto;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-7 {
  margin: 0;
  font-weight: 500;
  letter-spacing: 0;
  line-height: 40px;
  font-size: 32px;
  font-family: NotoSansThai-SemiBold,sans-serif;
  margin-bottom: 16px;
}

.emotion-8 {
  margin: 0;
  font-weight: 400;
  letter-spacing: 0;
  line-height: 32px;
  font-size: 24px;
  font-family: NotoSansThai-Regular,sans-serif;
  color: #757575;
  margin-bottom: 16px;
}

.emotion-9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-top: 8px;
}

.emotion-9>:not(style):not(style) {
  margin: 0;
}

.emotion-9>:not(style)~:not(style) {
  margin-top: 16px;
}

.emotion-10 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  min-width: 0;
  padding: 0;
  margin: 0;
  border: 0;
  vertical-align: top;
  width: 100%;
  background: #fff;
  border-radius: 8px;
  margin-top: 2em;
}

.emotion-11 {
  font-family: NotoSansThai-Regular,sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.4375em;
  color: rgba(0, 0, 0, 0.87);
  box-sizing: border-box;
  position: relative;
  cursor: text;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  position: relative;
  border-radius: 8px;
  padding-right: 14px;
}

.emotion-11.Mui-disabled {
  color: rgba(0, 0, 0, 0.38);
  cursor: default;
}

.emotion-11:hover .MuiOutlinedInput-notchedOutline {
  border-color: rgba(0, 0, 0, 0.87);
}

@media (hover: none) {
  .emotion-11:hover .MuiOutlinedInput-notchedOutline {
    border-color: rgba(0, 0, 0, 0.23);
  }
}

.emotion-11.Mui-focused .MuiOutlinedInput-notchedOutline {
  border-color: #1976d2;
  border-width: 2px;
}

.emotion-11.Mui-error .MuiOutlinedInput-notchedOutline {
  border-color: #d32f2f;
}

.emotion-11.Mui-disabled .MuiOutlinedInput-notchedOutline {
  border-color: rgba(0, 0, 0, 0.26);
}

.emotion-12 {
  font: inherit;
  letter-spacing: inherit;
  color: currentColor;
  padding: 4px 0 5px;
  border: 0;
  box-sizing: content-box;
  background: none;
  height: 1.4375em;
  margin: 0;
  -webkit-tap-highlight-color: transparent;
  display: block;
  min-width: 0;
  width: 100%;
  -webkit-animation-name: mui-auto-fill-cancel;
  animation-name: mui-auto-fill-cancel;
  -webkit-animation-duration: 10ms;
  animation-duration: 10ms;
  padding: 16.5px 14px;
  padding-right: 0;
}

.emotion-12::-webkit-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-12::-moz-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-12:-ms-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-12::-ms-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-12:focus {
  outline: 0;
}

.emotion-12:invalid {
  box-shadow: none;
}

.emotion-12::-webkit-search-decoration {
  -webkit-appearance: none;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-12::-webkit-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-12::-moz-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-12:-ms-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-12::-ms-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-12:focus::-webkit-input-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-12:focus::-moz-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-12:focus:-ms-input-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-12:focus::-ms-input-placeholder {
  opacity: 0.42;
}

.emotion-12.Mui-disabled {
  opacity: 1;
  -webkit-text-fill-color: rgba(0, 0, 0, 0.38);
}

.emotion-12:-webkit-autofill {
  -webkit-animation-duration: 5000s;
  animation-duration: 5000s;
  -webkit-animation-name: mui-auto-fill;
  animation-name: mui-auto-fill;
}

.emotion-12:-webkit-autofill {
  border-radius: inherit;
}

.emotion-13 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  text-align: center;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  font-size: 1.5rem;
  padding: 8px;
  border-radius: 50%;
  overflow: visible;
  color: rgba(0, 0, 0, 0.54);
  -webkit-transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-13::-moz-focus-inner {
  border-style: none;
}

.emotion-13.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-13 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-13:hover {
  background-color: rgba(0, 0, 0, 0.04);
}

@media (hover: none) {
  .emotion-13:hover {
    background-color: transparent;
  }
}

.emotion-13.Mui-disabled {
  background-color: transparent;
  color: rgba(0, 0, 0, 0.26);
}

.emotion-14 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
}

.emotion-15 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

.emotion-16 {
  text-align: left;
  position: absolute;
  bottom: 0;
  right: 0;
  top: -5px;
  left: 0;
  margin: 0;
  padding: 0 8px;
  pointer-events: none;
  border-radius: inherit;
  border-style: solid;
  border-width: 1px;
  overflow: hidden;
  min-width: 0%;
  border-color: rgba(0, 0, 0, 0.23);
}

.emotion-17 {
  float: unset;
  width: auto;
  overflow: hidden;
  padding: 0;
  line-height: 11px;
  -webkit-transition: width 150ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: width 150ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
}

.emotion-18 {
  margin: 0;
  font-weight: 400;
  line-height: 24px;
  letter-spacing: 0.5px;
  font-size: 16px;
  font-family: NotoSansThai-Regular,sans-serif;
  margin-top: 0px;
  color: #d32f2f;
  text-align: left;
}

.emotion-28 {
  margin: 0;
  font-weight: 400;
  line-height: 24px;
  letter-spacing: 0.5px;
  font-size: 16px;
  font-family: NotoSansThai-Regular,sans-serif;
  color: #d32f2f;
  text-align: left;
}

.emotion-30 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  color: #e0e0e0;
  margin-top: 8px;
}

.emotion-31 {
  margin: 0;
  font-weight: 500;
  line-height: 24px;
  letter-spacing: 0.5px;
  font-size: 16px;
  font-family: NotoSansThai-SemiBold,sans-serif;
}

.emotion-32 {
  margin: 0;
  font-weight: 400;
  line-height: 24px;
  letter-spacing: 0.5px;
  font-size: 16px;
  font-family: NotoSansThai-Regular,sans-serif;
  white-space: break-spaces;
}

.emotion-33 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: NotoSansThai-Regular,sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.75;
  text-transform: uppercase;
  min-width: 64px;
  padding: 6px 8px;
  border-radius: 8px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  color: #1976d2;
  width: 100%;
  height: 52px;
  margin: 2em 0 0;
  background-color: #eeeeee;
  color: #fff;
}

.emotion-33::-moz-focus-inner {
  border-style: none;
}

.emotion-33.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-33 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-33:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(25, 118, 210, 0.04);
}

@media (hover: none) {
  .emotion-33:hover {
    background-color: transparent;
  }
}

.emotion-33.Mui-disabled {
  color: rgba(0, 0, 0, 0.26);
}

.emotion-33:hover {
  background-color: #42a5f5;
  opacity: 0.9;
}

.emotion-34 {
  margin: 0;
  font-weight: 500;
  line-height: 24px;
  letter-spacing: 0.1px;
  font-size: 16px;
  font-family: NotoSansThai-SemiBold,sans-serif;
}

<div
    class="MuiStack-root emotion-0"
  >
    <div
      class="MuiStack-root emotion-1"
    >
      <img
        alt="login-background"
        src="/img.jpg"
      />
      <svg
        class="emotion-2"
      />
      <svg
        class="emotion-3"
      />
    </div>
    <div
      class="MuiStack-root emotion-4"
    >
      <div
        class="MuiStack-root emotion-5"
      >
        <div
          class="MuiStack-root emotion-6"
        >
          <h2
            class="MuiTypography-root MuiTypography-headlineLargeSemiBold emotion-7"
          >
            ตั้งค่ารหัสผ่านใหม่
          </h2>
          <h2
            class="MuiTypography-root MuiTypography-headlineSmall emotion-8"
          >
            กรอกรหัสผ่านใหม่เพื่อตั้งค่ารหัสผ่านใหม่
          </h2>
          <form>
            <div
              class="MuiStack-root emotion-9"
            >
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root emotion-10"
                data-testid="newPassword"
              >
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd emotion-11"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd emotion-12"
                    id="newPassword"
                    name="newPassword"
                    placeholder="ตั้งค่ารหัสผ่าน"
                    type="password"
                    value=""
                  />
                  <a>
                    <button
                      aria-label="Toggle show password"
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium emotion-13"
                      data-testid="button-toggle-show-new-password"
                      tabindex="0"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        aria-label="Hiding password"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-14"
                        data-testid="VisibilityOffIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7M2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2m4.31-.78 3.15 3.15.02-.16c0-1.66-1.34-3-3-3z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root emotion-15"
                      />
                    </button>
                  </a>
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline emotion-16"
                  >
                    <legend
                      class="emotion-17"
                    >
                      <span
                        class="notranslate"
                      >
                        ​
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
              <span
                class="MuiTypography-root MuiTypography-bodyLarge emotion-18"
                data-testid="error"
              />
            </div>
            <div
              class="MuiStack-root emotion-9"
            >
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root emotion-10"
                data-testid="confirmNewPassword"
              >
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth Mui-focused MuiInputBase-formControl MuiInputBase-adornedEnd emotion-11"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd emotion-12"
                    id="confirmNewPassword"
                    name="confirmNewPassword"
                    placeholder="ตั้งรหัสผ่านใหม่อีกครั้ง"
                    type="password"
                    value=""
                  />
                  <a>
                    <button
                      aria-label="Toggle show password"
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium emotion-13"
                      data-testid="button-toggle-show-confirm-password"
                      tabindex="0"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        aria-label="Hiding password"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-14"
                        data-testid="VisibilityOffIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7M2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2m4.31-.78 3.15 3.15.02-.16c0-1.66-1.34-3-3-3z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root emotion-15"
                      />
                    </button>
                  </a>
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline emotion-16"
                  >
                    <legend
                      class="emotion-17"
                    >
                      <span
                        class="notranslate"
                      >
                        ​
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
              <span
                class="MuiTypography-root MuiTypography-bodyLarge emotion-28"
                data-testid="error"
              />
            </div>
            <span
              class="MuiTypography-root MuiTypography-bodyLarge emotion-28"
            />
            <div
              class="MuiStack-root emotion-30"
            >
              <span
                class="MuiTypography-root MuiTypography-bodyLargeSemiBold emotion-31"
              >
                เงื่อนไขการตั้งรหัสผ่าน
              </span>
              <span
                class="MuiTypography-root MuiTypography-bodyLarge emotion-32"
              >
                 - มีอย่างน้อย 8 หลัก
 - มีตัวเลขอย่างน้อย 1 หลัก
 - ประกอบด้วยอักษรภาษาอังกฤษ
   - ตัวพิมพ์ใหญ่อย่างน้อย 1 ตัว
   - ตัวพิมพ์เล็กอย่างน้อย 1 ตัว
   - สัญลักษณ์อย่างน้อย 1 ตัว
              </span>
            </div>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary Mui-disabled MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary emotion-33"
              data-testid="button-set-new-password"
              disabled=""
              tabindex="-1"
              type="submit"
            >
              <span
                class="MuiTypography-root MuiTypography-labelExtraLargeSemiBold emotion-34"
              >
                ตั้งรหัสผ่านใหม่
              </span>
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
